### PR TITLE
fix(readiness): report host OS qualification

### LINKED ISSUE #11026: [All Platforms][CLI&UX] host probe records no host OS distribution or version, so an unqualified OS release is never warned about before install
## Description

`nemoclaw host probe` records no host operating-system distribution or version, so an unsupported or untested OS is never surfaced to the user before install or onboarding proceeds.

The JSON report contains **36 observations**. It records the platform family (`host.os.platform = linux`), the architecture, and whether the host is WSL, but there is no observation carrying the distribution name or its version. The only observation ids containing a version are `host.docker.cgroup_version`, `host.gpu.driver_version` and `gateway.version_drift`, none of which describe the host OS. No finding of any severity is emitted for the OS version either.

The practical consequence: a user on an OS the product has not qualified gets no early warning from the readiness surface, and the first signal is a failure later in install or onboarding. A generic Linux or WSL host is accepted regardless of the distribution release it is running.

**Platform scope:** Reproduced on DGX Station (Ubuntu 24.04.4 LTS, aarch64) and on a WSL2 host (Ubuntu 24.04.4 LTS, x86_64). Both report the same 36 observations with no OS version.
**Regression:** Unknown — earlier versions not tested for this.

## Environment

```text
Device:        DGX Station (aarch64) and a WSL2 x86_64 host
OS:            Ubuntu 24.04.4 LTS on both
Architecture:  aarch64 and x86_64
Node.js:       v22.23.2
npm:           10.9.8
Docker:        29.2.1 (Station), Docker Desktop 29.5.3 (WSL2)
OpenShell CLI: 0.0.106
NemoClaw:      v0.0.119
```

## Steps to Reproduce

1. On any supported Linux or WSL2 host, run `nemoclaw host probe`.
2. Run `nemoclaw host probe --json`.
3. Inspect the observation ids for one describing the host OS distribution or its version.

## Expected Result

The readiness report records the host operating system distribution and version, and emits a warning when the host runs a release the product has not qualified, so the user learns this before install or onboarding proceeds.

## Actual Result

```text
36 observations are returned. The ones carrying a version are:
  host.docker.cgroup_version
  host.gpu.driver_version
  gateway.version_drift

None describes the host OS. The OS-related observations are limited to:
  host.os.platform     = linux
  host.os.architecture = arm64 (or x64)
  host.os.wsl          = true or false

No finding mentions the operating system release.
```

## Logs

```text
DGX Station, Ubuntu 24.04.4 LTS aarch64:
  observation count 36; version-bearing ids: host.docker.cgroup_version, host.gpu.driver_version, gateway.version_drift

WSL2, Ubuntu 24.04.4 LTS x86_64:
  observation count 36; version-bearing ids: host.docker.cgroup_version, host.gpu.driver_version, gateway.version_drift
```

## Accepted implementation plan

- Decision: **Accept**.
- Reason: Host readiness must identify the Linux distribution and release before install or onboarding reaches a later platform failure.
- Placement: `main`, in host probe collection, local and pinned-SSH transport, readiness findings, CLI/JSON output, and the owning readiness documentation.
- Accountable maintainer: @rsliter.
- Delivery vehicle: PR #11780. Do not create a competing implementation.
- Outcome: Collect bounded `/etc/os-release` evidence, publish stable OS observations, and warn when qualification is unsupported or inconclusive.
- Acceptance:
  - Qualified Ubuntu evidence produces stable distribution and version observations without a warning.
  - Unsupported, missing, malformed, repeated-key, oversized, NUL, and carriage-return evidence produces the defined warning or inconclusive result without unsafe parsing.
  - Linux, WSL, local, and pinned-SSH paths have focused regression evidence.
  - The readiness documentation matches the CLI and JSON contracts.
- Exclusions: Do not turn the advisory into an install block. Do not claim qualification for untested releases.
- Stop condition: Continue through PR #11780 review and CI. Re-plan if review changes the qualification policy or supported release set.

### PR DESCRIPTION
**Pull request by cjagwani**
## Summary

Relays the reviewed change from #11291 through a same-repository branch so the protected NVIDIA CI can access the reviewed OpenShell SDK package. Deepak's source head remains preserved as an ancestor; this relay applies its exact 12-file patch to current `main`.

## Related Issue

Fixes #11026.

## Changes

- Preserve all nine verified, signed-off source commits from #11291.
- Collect bounded `/etc/os-release` identity for Linux and WSL hosts.
- Reject oversized, malformed, repeated-key, NUL, and carriage-return release evidence.
- Safely support the standard relative `/etc/os-release` symlink in local and pinned-SSH transports.
- Publish stable OS observations and warn before onboarding when release qualification is unsupported or inconclusive.
- Add focused regressions and update the system-readiness contract.

## Type of Change

- [ ] Code change (feature, bug fix, or refactor)
- [x] Code change with doc updates
- [ ] Doc only (prose changes, no code sample modifications)
- [ ] Doc only (includes code sample changes)

## Quality Gates

- [x] Tests added or updated for changed behavior
- [ ] Existing tests cover changed behavior — justification:
- [ ] Tests not applicable — justification:
- [x] Docs updated for user-facing behavior changes
- [ ] Docs not applicable — justification:
- [x] Sensitive paths changed (security, policy, credentials, preflight, onboarding, inference, runner, sandbox, or messaging)
- [x] Sensitive-path review completed or maintainer-approved waiver recorded — reviewer/approval link/justification: #11291 is approved; its exact source head passed 24/24 bounded review packets and the nine-category security review with no actionable finding.
- [ ] Non-success, skipped, or missing CI check accepted by maintainer — check name, approval link, and follow-up issue:

## Verification

- [x] PR description includes a `Signed-off-by:` line and every commit appears as `Verified` in GitHub
- [x] Normal `pre-commit`, `commit-msg`, and `pre-push` hooks passed, or `npm run check:diff` passed when hooks were skipped or unavailable
- [x] Targeted behavior tests pass for the current change set, or tests are marked not applicable above — 200 focused tests passed on the reviewed source head.
- [ ] Applicable broad gate passed — fresh protected CI and the staged E2E pull-request gate are running on the current relay head.
- [x] Quality Gates section completed with required justifications or waivers
- [x] No secrets, API keys, or credentials committed
- [ ] `npm run docs` builds without warnings (doc changes only) — source validation reported no errors and only existing warnings.
- [x] Doc pages follow the [style guide](https://github.com/NVIDIA/NemoClaw/blob/main/docs/CONTRIBUTING.md) (doc changes only)
- [ ] New doc pages include SPDX header and frontmatter (new pages only)

## Relay provenance

- Source PR: #11291
- Source head: `671697e7db29c6bf9d70cc207c56ea1211eefbf1`
- Relay head: `44312709f0cab801cfad1e9dec98a98a6330ac95`
- Current main parent: `97745a7ad9649f851704493e4b670b3674f875aa`
- The relay was updated by signed fast-forward merge commits; no contributor commit or signature was rewritten.

---
Signed-off-by: Deepak Jain <deepujain@gmail.com>
Signed-off-by: Charan Jagwani <cjagwani@nvidia.com>

### CODE REVIEW

File: src/lib/inference/serving/managed-cluster-discovery.test.ts [hunk 1]
```diff
@@ -943,6 +943,87 @@ describe("production pinned peer transport", () => {
...
+    const deps = createManagedClusterDiscoveryDeps(() => ({ status: 0, stdout: "", stderr: "" }));
+
+    deps.createReadiness(
+      host("local"),
+      candidate,
+      { nemoclawVersion: "0.1.0", sourceRevision: SOURCE_REVISION },
+      NOW,
+    );
+
+    expect(readOsRelease).toHaveBeenCalledWith("/etc/os-release", "/usr/lib/os-release", 4096);
```
[the remark this answers is not in the document]
Comment (cjagwani) [reply src/lib/inference/serving/managed-cluster-discovery.test.ts:1001]: Verified against the current head. This is a useful additional integration-test suggestion, but it is not a merge blocker: the production filesystem test executes the bounded reader for the standard relative symlink, arbitrary-link rejection, and missing-primary fallback; platform-qualification tests cover valid, NUL, carriage-return, malformed, and duplicate-key parsing; host tests assert the resulting public observations/findings; and the canonical wiring test proves createReadiness selects readOsRelease with the 4096-byte limit. The requested four-case cross-layer matrix would duplicate those already-covered contracts and does not identify incorrect production behavior. The exact focused suite (200 tests) and the full protected relay CI both pass.

**Comment by cjagwani** [comment]
@prekshivyas This relay preserves the exact #11291 source head you approved and applies that patch to current main. Protected OpenShell SDK packaging and all required checks pass. Please approve the relay provenance when you have a moment; no additional code review should be needed.

**Review by rsliter — APPROVED** [review]
Reviewed commit 6d8d5616de75d116f6cca400b2a938809ae4098b. The OS release reader is bounded to 4,096 bytes, handles the standard symlink and fallback paths, rejects unrelated symlink targets, and projects the stable Linux and WSL observations. Focused regression tests, the documentation build, full PR validation, all 63 current checks, the live generic-GPU llama.cpp qualification, and the Review Advisor no-blocker gate pass. No blocking issues found.
